deb/debian/preinst

```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+# preinst script for stackdriver-agent
+#
+# see: dh_installdeb(1)
+
+set -e
+
+# summary of how this script can be called:
+#        * <preinst> `install'
+#        * <preinst> `install' <old-version> <new-version>
+#        * <new-preinst> `upgrade' <old-version> <new-version>
+#        * <old-preinst> `abort-upgrade' <old-version>
+# for details, see http://www.debian.org/doc/debian-policy/ or
+# the debian-policy package
+
+. /usr/share/debconf/confmodule
+
+OPT_COLLECTD="/opt/stackdriver/collectd/etc"
+ARCHIVE_DIR="/etc/stackdriver/archive"
+
+case "$1" in
+    install)
+        ;;
+
+    upgrade)
+      if [ -f "$OPT_COLLECTD/collectd.conf.tmpl" ]; then
+        if [ ! -d "$ARCHIVE_DIR" ]; then
```

Comment:
[Optional] This could just be:
```sh
      [ -d "$ARCHIVE_DIR" ] || mkdir -p "$ARCHIVE_DIR"
      cp "$OPT_COLLECTD/collectd.conf.tmpl" \
        "$ARCHIVE_DIR/collectd.conf.tmpl.old"
```

Reply:
Done.